### DISCUSSION TITLE
Pinebook pro (linux-aarch64-32)

### DISCUSSION THREAD

**Discussion by sulivanShu** [desyncr/mobile-node#97] [Q&A] [answered]
This is a related topic: I can't get Freenet to work on the Pinebook pro, which is an ARM platform.

Since a command line is better than a long speech:

```
$ uname --all
Linux manjaro 5.13.7-1-MANJARO-ARM #1 SMP Sat Jul 31 07:25:28 UTC 2021 aarch64 GNU/Linux
$ java --version
openjdk 16.0.2 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7)
OpenJDK 64-Bit Server VM (build 16.0.2+7, mixed mode)
$ ./run.sh start
Your java executable at /bin/java seems suitable
Unable to locate any of the following binaries:
  /home/regivanx/Freenet/./bin/wrapper-linux-aarch64-32
  /home/regivanx/Freenet/./bin/wrapper
Starting Freenet 0.7...

Let's start the node without the wrapper, you'll have to daemonize it yourself.
Exception in thread "main" java.lang.NoClassDefFoundError: com/sun/jna/Platform
        at freenet.support.ProcessPriority.enterBackgroundMode(ProcessPriority.java:63)
        at freenet.node.NodeStarter.main(NodeStarter.java:264)
Caused by: java.lang.ClassNotFoundException: com.sun.jna.Platform
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
        ... 2 more
```

Maybe it's related to the difficulty of getting Freenet to work on arm? Do you have any advice for me to get Freenet running on my computer?

**Comment by desyncr** [comment]
Thanks for reporting. From the error message I can see that JNA-platform [package](https://github.com/freenet-mobile/app/blob/master/app/build.gradle#L59) fails to load in this architecture. I believe JNA [should be available](https://github.com/java-native-access/jna/tree/master/lib/native), but otherwise there's little to do from ourside.

Edit: I just realized you're talking about the Desktop version, not the mobile one.

**Comment by sulivanShu** [comment, marked as answer]
Hi! The matter has been discussed here: [Freenet on Pinebook pro](https://forum.pine64.org/showthread.php?tid=15555&pid=104081). The problem was solved by a hack.

**Reply by desyncr** [reply]
Thanks for reporting back. Appreciated!
